nixos/modules/services/networking/nordvpn.nix

```diff
@@ -0,0 +1,66 @@
+{
+  lib,
+  config,
+  pkgs,
+  proc-ps,
+  ...
+}:
+
+let
+  cfg = config.services.nordvpn;
+  nordvpnPkg = cfg.package;
+in
+{
+  options.services.nordvpn = {
+
+    enable = lib.mkEnableOption "Enable NordVPN";
+
+    package = lib.mkPackageOption pkgs "nordvpn" { };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.tmpfiles.rules = [
+      # nordvpnd relies on hardcorded path to the following
+      "L /usr/lib/nordvpn - - - - ${nordvpnPkg}/lib/nordvpn"
+      # this fold will contain the socket used between nordvpnd and nordvpn
+      "D /run/nordvpn 0770 root nordvpn"
+    ];
+
+    users.groups.nordvpn = { };
```

Comment:
Is it needed ? Can't we use DynamicUser ?

Reply:
Have you tried removing that and using systemd's `DynamicUser` directive? Have a look at other NixOS services for some inspiration.